wardrobe/timestamp/src/lib.rs

```diff
@@ -167,45 +159,32 @@ impl<T: TimestampConfig + 'static, V: Verifier + From<UpForGrabs>> ConstraintChe
         // In Tuxedo, inherent transaction types are identified explicitly, and the tuxedo
         // core can make this check automatically.
 
-        // Make sure the first output is a new best timestamp
+        // Make sure there are no inputs. Setting a new timestamp does not consume anything.
         ensure!(
-            !output_data.is_empty(),
-            Self::Error::MissingNewBestTimestamp
+            input_data.is_empty(),
+            Self::Error::InputsWhileSettingTimestamp
         );
-        let new_best = output_data[0]
-            .payload
-            .extract::<BestTimestamp>()
-            .map_err(|_| Self::Error::BadlyTyped)?
-            .0;
 
-        // Make sure the second output is a new noted timestamp
-        ensure!(
-            output_data.len() >= 2,
-            Self::Error::MissingNewNotedTimestamp
-        );
-        let new_noted = output_data[1]
+        // Make sure the only output is a new best timestamp
+        ensure!(!output_data.is_empty(), Self::Error::MissingNewTimestamp);
+        let new_timestamp = output_data[0]
             .payload
-            .extract::<NotedTimestamp>()
-            .map_err(|_| Self::Error::BadlyTyped)?
-            .0;
-
-        // Make sure there are no extra outputs
+            .extract::<Timestamp>()
+            .map_err(|_| Self::Error::BadlyTyped)?;
         ensure!(
-            output_data.len() == 2,
+            output_data.len() >= 2,
```

Comment:
I thought the output of the inherent was supposed to have length 1.
Maybe we can simplify the check, and instead of having two ensure!, we could only check if the length is 1.

Reply:
fixed part 1 in 10213120bb46a1a3e6cc48d25907d14540f292b4

Reply:
Yeah, good call. I had this wrong.